source/configuration_manual/mail_location/obox/amazon_s3.rst

```diff
@@ -35,20 +42,36 @@ partitions = this would theoretically provide you up to: ``~896,000
 PUT/POST/DELETE TPS and 1,408,000`` GET TPS if requests are distributed evenly
 across the partitions).
 
+Configuration
+^^^^^^^^^^^^^
+
 .. code-block:: none
 
   plugin {
-    obox_fs = fscache:1G:/var/cache/mails:compress:gz:6:s3: https://ACCESSKEY:SECRET@BUCKETNAME.s3.amazonaws.com/
-    obox_index_fs = compress:gz:6:s3: https://ACCESSKEY:SECRET@BUCKETNAME.s3.amazonaws.com/
+    obox_fs = fscache:1G:/var/cache/mails:compress:gz:6:aws-s3:https://ACCESSKEY:SECRET@BUCKETNAME.s3.amazonaws.com/
+    obox_index_fs = compress:gz:6:aws-s3:https://ACCESSKEY:SECRET@BUCKETNAME.s3.amazonaws.com/
   }
 
+.. Note::
+        If the version you are using does not support the ``aws-s3`` scheme,
+        try using the more generic ``s3`` scheme.
+
+        .. versionadded:: 2.3.10
+
 Get ACCESSKEY and SECRET from `www.aws.amazon.com <https://aws.amazon.com/>`_
 ``-> My account -> Security credentials -> Access`` credentials. Create the
 ``BUCKETNAME`` from ``AWS Management Console -> S3 -> Create Bucket``.
 
 If the ``ACCESSKEY`` or ``SECRET`` contains any special characters, they can be
 %hex-encoded.
 
+If the ``aws-3`` scheme is used dovecot defaults to append the following url
+parameters, refer to :ref:`http_based_object_storages` for details.
+
+.. code-block:: none
+
+  auth_protocol=iam&addhdrvar=x-amz-security-token:%{auth:token}&loghdr=x-amz-request-id&loghdr=x-amz-id-2
```

Comment:
Oh right, this brings to my mind another point: The aws-s3 currently requires IAM, meaning it only works in AWS's VMs. It would be nice if it was possible to disable IAM by using `aws-s3:http://.../?auth_protocol=` but because this string is now appended it overwrites the auth_protocol.. Maybe it should be prepended after the `?` instead so auth_protocol can be overwritten?

And the documentation should explain about using S3 in AWS VMs vs externally..

Reply:
Although I'm not sure how x-amz-security-token works in that situation. I guess if auth:token expands to empty it could avoid adding the header (not sure if it already does). But the loghdrs are the main nice and easy thing about aws-s3.